code/modules/atmospherics/FEA_system.dm

```diff
@@ -1,322 +1,280 @@
+
+/// Updates cached space sample if need be.
+// Returns: New space sample.
+/datum/controller/air_system/proc/update_space_sample()
+	if (!space_sample || !(space_sample.turf_flags & CAN_BE_SPACE_SAMPLE))
+		space_sample = locate(/turf/space)
+	return space_sample
+
+/datum/controller/air_system/proc/setup(datum/controller/process/air_system/controller)
+	parent_controller = controller
+
+	#if SKIP_FEA_SETUP == 1
+	return
+	#else
+
+	boutput(world, "<span class='alert'>Processing Geometry...</span>")
+
+	var/start_time = world.timeofday
+
+	for(var/turf/simulated/S in world)
+		if(!S.gas_impermeable && !S.parent)
+			assemble_group_turf(S)
+		S.update_air_properties()
+
+	boutput(world, "<span class='alert'>Geometry processed in [(world.timeofday-start_time)/10] seconds!</span>")
+	#endif
+
+/// Collects turfs into groups.
+/datum/controller/air_system/proc/assemble_group_turf(turf/simulated/base)
+	set waitfor = 0
+	var/list/turf/simulated/members = list(base) // Confirmed group members
+	var/list/turf/simulated/possible_members = list(base) // Possible places for group expansion
+	var/list/turf/simulated/possible_borders
+	var/list/turf/simulated/possible_space_borders
+	var/possible_space_length = 0
+
+	while(length(possible_members)) //Keep expanding, looking for new members
+		for(var/turf/simulated/test as anything in possible_members)
+			test.length_space_border = 0
+			for(var/direction in cardinal)
+				var/turf/T = get_step(test,direction)
+				if(T && !(T in members) && test.gas_cross(T))
+					if(T.turf_flags & IS_TYPE_SIMULATED)
+						if(!T:parent)
+							possible_members += T
+							members += T
+						else
+							LAZYLISTINIT(possible_borders)
+							possible_borders |= test
+					else if(istype(T, /turf/space) && !istype(T, /turf/space/fluid))
+						LAZYLISTINIT(possible_space_borders)
+						possible_space_borders |= test
+						test.length_space_border++
+
+			if(test.length_space_border > 0)
+				possible_space_length += test.length_space_border
+			possible_members -= test
+
+	if(length(members) > 1)
+		var/datum/air_group/group = new
+		if(possible_borders && length(possible_borders))
+			group.borders = possible_borders
+		if(possible_space_borders && length(possible_space_borders))
+			group.space_borders = possible_space_borders
+			group.length_space_border = possible_space_length
+
+		for(var/turf/simulated/test as anything in members)
+			test.parent = group
+			test.processing = FALSE
+			active_singletons -= test
+
+			test.dist_to_space = null
+			var/dist
+			for(var/turf/simulated/possible as anything in possible_space_borders)
+				if (possible == test)
+					test.dist_to_space = 1
+					break
+				dist = GET_DIST(possible, test)
+				if (!test.dist_to_space || (dist < test.dist_to_space))
+					test.dist_to_space = dist
+
+		// Allow groups to determine if group processing is applicable after FEA setup
+		if(current_cycle) group.group_processing = FALSE
```

Comment:
i know this works fine but the lack of indentation on lines 163 and 164 confused me for a second. Maybe put `group.group_processing = FALSE` into another line for readability.